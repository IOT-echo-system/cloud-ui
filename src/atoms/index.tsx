export * from './StyledComponents'
export * from './FormInput'
export * from './ToastWrapper'
export * from './Loader'
