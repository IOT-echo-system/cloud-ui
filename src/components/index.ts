export {default as Layout} from './Layout'
export * from './Layout'

export {default as SiteWrapper} from './SiteWrapper'
export * from './SiteWrapper'

export * from './atom'
