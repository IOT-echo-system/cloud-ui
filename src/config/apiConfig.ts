export const apiConfig = {
  baseUrl: 'http://localhost:3001/api',
  auth: {
    baseUrl: '/auth',
    signUp: '/sign-up',
    login: '/login',
    validate: '/validate',
    generateOTP: '/generate-otp',
    verifyOTP: '/verify-otp',
    resetPassword: '/reset-password',
    updateToken: '/update-token',
    userDetails: '/user-details',
    logout: '/logout'
  },
  project: {
    baseUrl: '/projects',
    projects: '',
    projectsWithRoles: '/project-with-roles',
    project: '/{projectId}/role/{roleId}',
    projectDetails: '/project-details'
  },
  board: {
    baseUrl: '/boards',
    boards: '',
    board: '/{boardId}',
    updateBoardName: '/{boardId}/name',
    secretKey: '/{boardId}/secret-key'
  },
  widget: {
    baseUrl: '/widgets',
    widgets: ''
  }
} as const
