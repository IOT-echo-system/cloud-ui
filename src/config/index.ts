export const Config = {
  BACKEND_BASE_URL: '/api',
  SIGN_UP_PATH: '/authentications/sign-up',
  LOGIN_PATH: '/authentications/login',
  HOME_PAGE_PATH: '/'
} as const
