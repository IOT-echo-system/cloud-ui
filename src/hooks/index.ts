export {default as useForm} from './useForm'
export * from './useForm'

export * from './useStore'
export * from './useCountDownTimer'
export {default as useCountDownTimer} from './useCountDownTimer'
