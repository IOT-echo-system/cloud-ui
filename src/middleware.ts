import {NextResponse} from 'next/server'

export const middleware = (request: Request): NextResponse => {
  // eslint-disable-next-line no-process-env
  if (process.env.NODE_ENV === 'production') {
    return NextResponse.next()
  }

  const url = new URL(request.url)
  if (url.pathname.startsWith('/api')) {
    return NextResponse.rewrite('http://192.168.1.102:9000' + url.pathname)
  }

  return NextResponse.next()
}
