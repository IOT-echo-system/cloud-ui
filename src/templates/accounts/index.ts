export * from './signup/SignUp'
export * from './login/Login'
