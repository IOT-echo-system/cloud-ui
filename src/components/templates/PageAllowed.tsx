import React, {useEffect, useState} from 'react'
import {useSelector} from '../../hooks'
import {useRouter} from 'next/router'
import {Config} from '../../config'
import {Loader} from '../atoms'
import {PolicyUtils} from '../../utils/policyUtils'

type PageAllowedPropsType<T extends Record<string, unknown>> = {
  policy: string
  Component: React.ComponentType<T>
} & T

export const PageAllowed = <T extends Record<string, unknown>>(props: PageAllowedPropsType<T>): React.JSX.Element => {
  const {policy, Component} = props
  const {policies} = useSelector(state => state.user)
  const [validated, setValidated] = useState(false)
  const router = useRouter()

  useEffect(() => {
    const isAllowed = PolicyUtils.isValid(policies, policy)
    if (!isAllowed) {
      router.push(Config.HOME_PAGE_PATH).catch(() => ({}))
    }
    setValidated(true)
  }, [])

  if (!validated) {
    return <Loader loadingText={'Loading...'} />
  }

  return <Component {...props} />
}
