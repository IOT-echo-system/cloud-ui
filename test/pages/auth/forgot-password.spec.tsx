import {render} from '@testing-library/react'
import ForgotPasswordPage from '../../../src/pages/auth/forgot-password'
import {ResetPassword, VerifyOtp} from '../../../src/templates/accounts'
import React from 'react'
import {Config} from '../../../src/config'

jest.mock('../../../src/templates/accounts/verifyOtp/VerifyOtp')
jest.mock('../../../src/templates/accounts/resetPassword/ResetPassword')

jest.mock('react', (): typeof React => ({
  ...jest.requireActual('react'),
  useState: jest.fn()
}))

describe('Forgot password page test', () => {
  const mockUseState = React.useState as jest.Mock
  beforeEach(jest.clearAllMocks)
  afterEach(jest.clearAllMocks)

  it('should match with the snapshot', () => {
    mockUseState.mockReturnValue([false, jest.fn()])

    const {container} = render(<ForgotPasswordPage />)

    expect(container).toMatchSnapshot()
    expect(VerifyOtp).toHaveBeenCalledTimes(1)
    expect(VerifyOtp).toHaveBeenCalledWith({setOtpVerified: expect.any(Function), title: 'Forgot password'}, {})
    expect(ResetPassword).not.toHaveBeenCalled()
  })

  it('should match with the snapshot if OTP verified', () => {
    mockUseState.mockReturnValue([true, jest.fn()])

    const {container} = render(<ForgotPasswordPage />)

    expect(container).toMatchSnapshot()

    expect(ResetPassword).toHaveBeenCalledTimes(1)
    expect(ResetPassword).toHaveBeenCalledWith(
      {
        redirectTo: Config.LOGIN_PAGE_PATH,
        title: 'Change password',
        withOldPassword: false
      },
      {}
    )
    expect(VerifyOtp).not.toHaveBeenCalled()
  })
})
