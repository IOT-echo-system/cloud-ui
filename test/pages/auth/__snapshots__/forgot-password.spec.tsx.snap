// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot password page test should match with the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1edlbwa-MuiStack-root"
  >
    <div
      class="MuiStack-root css-kd7he9-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Already have an account?
        </p>
        <a
          class="css-9x8bsw"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Forgot password page test should match with the snapshot if OTP verified 1`] = `
<div>
  <div
    class="MuiStack-root css-1edlbwa-MuiStack-root"
  >
    <div
      class="MuiStack-root css-kd7he9-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Already have an account?
        </p>
        <a
          class="css-9x8bsw"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Forgot password page test should match with the snapshot if OTP verified for mobile 1`] = `
<div>
  <div
    class="MuiStack-root css-1edlbwa-MuiStack-root"
  >
    <div
      class="MuiStack-root css-1yxcwai-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Already have an account?
        </p>
        <a
          class="css-9x8bsw"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;
